include/zephyr/net/wifi_mgmt.h

```diff
@@ -1010,6 +1016,24 @@ struct wifi_dpp_params {
 	};
 };
 
+#define WIFI_WPS_PIN_MAX_LEN 8
+
+/** Get/Set operation for WPS pin*/
+enum wifi_wps_pin_op {
+	/** Get operation */
+	WIFI_WPS_PIN_GET = 0,
+	/** Set operation */
+	WIFI_WPS_PIN_SET = 1,
+};
+
+/** Wi-Fi wps pin setup */
+struct wifi_wps_pin_params {
+	/** wps pin operation */
+	enum wifi_wps_pin_op oper;
+	/** pin value*/
+	char pin[WIFI_WPS_PIN_MAX_LEN + 1];
+};
```

Comment:
An alternative is

```
struct wifi_wps_config_params {
  enum wifi_wps_oper oper;
  union {
    enum wifi_wps_pin_op oper;
    char pin[WIFI_WPS_PIN_MAX_LEN + 1];
  } pin_params;
};
```

We can even move the `pin_cfg` above to support, `WPS_PBC, WPS_PIN_SET, WPS_PIN_GET`.

Reply:
Updated as comment